lms/djangoapps/mobile_api/utils.py

```diff
@@ -30,12 +34,22 @@ def _wrapper(self, request, *args, **kwargs):
             """
             course_id = CourseKey.from_string(kwargs.pop('course_id'))
             with modulestore().bulk_operations(course_id):
-                course = get_course_with_access(
-                    request.user,
-                    'load_mobile' if verify_enrolled else 'load_mobile_no_enrollment_check',
-                    course_id,
-                    depth=depth
-                )
+                try:
+                    course = get_course_with_access(
+                        request.user,
+                        'load_mobile' if verify_enrolled else 'load_mobile_no_enrollment_check',
+                        course_id,
+                        depth=depth
+                    )
+                except Http404 as error:
+                    if any_unfulfilled_milestones(course_id, request.user.id):
```

Comment:
Note: this is redundant code but in order to distinguish between a 404 and milestone error, this was the cleanest way to do it since `has_access` in `get_course_with_access` returned only a boolean. Will add a comment.